### DISCUSSION TITLE
Sharing KeyCloak SSO session with clients placed inside iframe

### DISCUSSION THREAD

**Discussion by anatoly-spb** [keycloak/keycloak#20743] [Q&A]
Hello! 

I have two KeyCloak clients app1, app2 in my realm related to my two web applications http://mysite/app1 and http://mysite/app2.  

I want to show app2 inside iframe in app1, but when I login to app1 and redirect to app1 my application app2 does not see SSO session and redirect to KeyCloak login page:

![image](https://github.com/keycloak/keycloak/assets/13556403/1dc057ea-8b1e-4884-b322-ff2b0c415b89)

If I put app2 into anchor and click on it, everything is ok, app2 open without output login prompt. 

What can be wrong?

Thanks in advance,
Anatoly Shirokov

**Comment by nmanthena18** [comment]
@anatoly-spb - did you got any solution on this

**Comment by justin-tay** [comment]
This is likely due to cookie same site restrictions.

If the web application is at http://mysite and Keycloak is at http://keycloak:8080 then these are on different sites. In order for the browser to send the cookie to Keycloak, since it is in the inline frame of http://mysite, the cookie needs to be marked as `SameSite=None`. However `SameSite=None` also requires that the cookie be marked as `Secure` which means a https connection is needed with Keycloak or the browser rejects the cookie, which I assume is not the case and you are connecting to Keycloak using http. In that case the browser will not send the cookie to Keycloak. You can open the browser developer console using F12 find the network tab and look at the requests to Keycloak and see if the cookies are being blocked.

Since this is likely just a dev setup, you probably have the following options
- Configure Keycloak to run with TLS by generating a key and cert and starting with `--https-certificate-file=keycloak.crt --https-certificate-key-file=keycloak.key` and have it running on https://keycloak:8443 (or)
- Place Keycloak and your web application on the same site eg. http://mysite.home.arpa and http://keycloak.home.arpa by modifying your hosts file
